tests/nectarstorage/test_sqlite.py

```
@@ -34,3 +34,47 @@ def test_initialdata(self):
+        from unittest.mock import patch
+
+        original_connect = sqlite3.connect
+
+        def mock_connect(database, *args, **kwargs):
+            # If it's a standard path (not a memory URI), raise OperationalError
+            if not database.startswith("file:"):
+                raise sqlite3.OperationalError("unable to open database file")
+            return original_connect(database, *args, **kwargs)
+
+        with patch("sqlite3.connect", mock_connect):
+            store = MyStore(data_dir="/tmp/test_nonexistent_writable")
```

Ruff (0.15.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/test_nonexistent_writable"

(S108)